### PR TITLE
WIP: Added TestContainersContext to control multiple containers lifecycle

### PR DESCRIPTION
**Pull request by LMnet**
DONT MERGE WORK IN PROGRESS

### CODE REVIEW

File: src/main/scala/com/dimafeng/testcontainers/TestContainer.scala [hunk 1]
```diff
@@ -16,32 +16,32 @@ import scala.concurrent.Future
 trait ForEachTestContainer extends SuiteMixin {
   self: Suite =>
 
-  val container: Container
```
Comment (dimafeng) [inline src/main/scala/com/dimafeng/testcontainers/TestContainer.scala]: I think, this filed should stay and be annotated as deprecated to make it compatible with `override` in `override val container = ...`

File: src/main/scala/com/dimafeng/testcontainers/TestContainersContext.scala [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  def add(container: Container): Unit
+  def startAll()(implicit description: Description): Unit
+  def finishAll()(implicit description: Description): Unit
+  def failAll(e: Throwable)(implicit description: Description): Unit
+  def succeedAll()(implicit description: Description): Unit
+}
+
+object TestContainersContext {
+
+  class Empty extends TestContainersContext {
```
Comment (dimafeng) [inline src/main/scala/com/dimafeng/testcontainers/TestContainersContext.scala]: wouldn't it be convenient to have `Empty` and `Default` as objects?

File: src/main/scala/com/dimafeng/testcontainers/TestContainersContext.scala [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    def finishAll()(implicit description: Description): Unit = {}
+    def failAll(e: Throwable)(implicit description: Description): Unit = {}
+    def succeedAll()(implicit description: Description): Unit = {}
+  }
+
+  class Default extends TestContainersContext {
+
+    import scala.collection.JavaConverters._
+
+    private val containers = new CopyOnWriteArrayList[Container]()
```
Comment (dimafeng) [inline src/main/scala/com/dimafeng/testcontainers/TestContainersContext.scala]: Just thinking, maybe mutable scala collections would fit better (e.g. `mutable.ArrayBuffer`) to get rid of `asScala` conversions and have more scala-ish `add` - `+=`

**Comment by LMnet** [comment]
This scheme will not work with the `lazy val`s. So, I will stay with the current solutions with `MultipleContainers` and `LazyContainer`

**Pull request closed without merging**